src/api/student.js

```diff
@@ -15,4 +16,26 @@ app.get('/', (req, res) => {
     res.successJson(student);
 });
 
+app.post('/', (req, res) => {
+    let workbook = xlsx.read(req.body.data, {type: 'binary'});
+    let studentJSON = xlsx.utils.sheet_to_json(workbook.Sheets[workbook.SheetNames[0]], {range:1});
+    for (let key in studentJSON) {
+        if (studentJSON.hasOwnProperty(key)) {
+            let s = studentJSON[key];
+            let student = {
+                id: s['Student ID'],
+                name: s['Student'],
+                status: s['Status'],
+                email: s['E-mail'],
+                major: s['Major']
+            };
+            if (StudentStore.isCurrentStudent(student) && StudentStore.isNewStudent(student)) {
+                addAction('NEW_STUDENT', student);
```

Comment:
Also I tried to do a quick refactor to make it send one action with an array of all students and that didn't work out too well. There are more tasks that need to be finished so this may be something we need to accept. Again refer to the comment above about race conditions, it will be very rare.